test/explorer/series/duration_test.exs

```diff
@@ -0,0 +1,368 @@
+      assert ms |> Series.cast({:duration, :nanosecond}) |> Series.all_equal(ns)
+      assert us |> Series.cast({:duration, :millisecond}) |> Series.all_equal(ms)
+      assert us |> Series.cast({:duration, :nanosecond}) |> Series.all_equal(ns)
+      assert ns |> Series.cast({:duration, :millisecond}) |> Series.all_equal(ms)
+      assert ns |> Series.cast({:duration, :microsecond}) |> Series.all_equal(us)
+    end
+
+    test "can convert to a list and back without needing the `dtype` option" do
+      ms = Series.from_list([1], dtype: {:duration, :millisecond})
+      us = Series.from_list([1_000], dtype: {:duration, :microsecond})
+      ns = Series.from_list([1_000_000], dtype: {:duration, :nanosecond})
+
+      assert ms |> Series.to_list() |> Series.from_list() |> Series.all_equal(ms)
+      assert us |> Series.to_list() |> Series.from_list() |> Series.all_equal(us)
+      assert ns |> Series.to_list() |> Series.from_list() |> Series.all_equal(ns)
+    end
+  end
+
+  describe "io" do
+    test "series to and from binary" do
+      for precision <- [:millisecond, :microsecond, :nanosecond] do
+        dtype = {:duration, precision}
+        durations = Series.from_list([100, 101], dtype: dtype)
+
+        [binary] = Series.to_iovec(durations)
+        from_binary = Series.from_binary(binary, dtype)
+
+        assert durations.dtype == from_binary.dtype
+        assert Series.to_list(durations) == Series.to_list(from_binary)
+      end
+    end
+
+    test "duration structs to_string similarly to polars" do
+      strings = [
+        "1ms",
+        "10ms",
+        "100ms",
+        "1s",
+        "10s",
+        "1m 40s",
+        "16m 40s",
+        "2h 46m 40s",
+        "1d 3h 46m 40s",
+        "11d 13h 46m 40s",
+        "115d 17h 46m 40s",
+        # Like polars, the maximum unit is days so we don't show years.
+        "1157d 9h 46m 40s"
+      ]
+
+      for {string, power} <- Enum.with_index(strings) do
+        assert to_string(%Duration{value: 10 ** power, precision: :millisecond}) == string
+      end
+    end
+
+    test "duration structs inspect as \"Duration[*]\"" do
+      assert inspect(%Duration{value: 1, precision: :millisecond}) == "Duration[1ms]"
+    end
+
+    test "in a series, equal values are displayed the same regardless of precision" do
+      ms = Series.from_list([1], dtype: {:duration, :millisecond})
+      us = Series.from_list([1_000], dtype: {:duration, :microsecond})
+      ns = Series.from_list([1_000_000], dtype: {:duration, :nanosecond})
+
+      # Each series displays its values as "[1ms]" as well as the correct precision.
+      assert inspect(ms) == """
+             #Explorer.Series<
+               Polars[1]
+               duration[ms] [1ms]
+             >\
+             """
+
+      assert inspect(us) == """
+             #Explorer.Series<
+               Polars[1]
+               duration[μs] [1ms]
+             >\
+             """
```

Comment:
I took a pass at implementing the string-related protocols while I was adding the duration struct. Not having them broke one of my tests (plus I was tired of counting 0's). Feel free to 👍 / 👎 on the specifics. I mostly tried to copy what polars was doing. But for `inspect` I went with:

```elixir
inspect(%Explorer.Duration{value: 1, precision: :millisecond})
#=> "Duration[1ms]"
```

I don't feel strongly about that choice, I just went with something.